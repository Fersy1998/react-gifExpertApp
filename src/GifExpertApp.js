//import React, { useState, useEffect } from 'react'
import React, { useState} from 'react'
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({deFaultCategories=[]}) => {
  //const categorias=['Sailor Moon', 'Cowboy Bebop', 'Kimagure Orange Road 1987'];

  const [categories, setCategories] = useState(deFaultCategories);
  //Agrega una nueva categorÃ­a
  
  /*
  const handledadd=()=>{
    setCategories(categorias=>[...categorias, "Air"]);
  }
  */
  return (
    <><h2>GifExpertApp</h2>
    {//La appa de abajo manda una referecia al hijo (AddCategory) de la constante setCategories
    }
    <AddCategory setCategories={setCategories}/>
    <hr />
    {
    //<button onClick={handledadd}>Agregar</button>
    
    }
    
     
      
      <div >{categories.map(category=> <GifGrid category={category} key={category}/>)}</div> 
    
    
    
    </>
    
  )
}

GifExpertApp.propTypes = {}

export default GifExpertApp