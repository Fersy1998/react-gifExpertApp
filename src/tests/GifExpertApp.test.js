import React from 'react';
import '@testing-library/jest-dom';
import GifExpertApp from '../GifExpertApp';
import {shallow} from 'enzyme';

describe('pruebas en GifExpertApp.js', ()=>{
    test('should match the snapshot', () => { 
        //const categorias=['Sailor Moon', 'Cowboy Bebop', 'Kimagure Orange Road 1987']
        const wrapper=shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
        
    })
    test('should match the snapshot', () => { 
        const categorias=['Sailor Moon', 'Cowboy Bebop', 'Kimagure Orange Road 1987']
        const wrapper=shallow(<GifExpertApp deFaultCategories={categorias}/>);
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);
        
    })

})