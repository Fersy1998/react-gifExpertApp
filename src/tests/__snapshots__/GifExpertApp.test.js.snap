// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifExpertApp.js should match the snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div />
</Fragment>
`;
